src/db.rs

```diff
@@ -41,41 +74,129 @@ impl AgateOptions {
         self
     }
 
-    pub fn wal_path<P: Into<PathBuf>>(&mut self, p: P) -> &mut AgateOptions {
-        self.wal_path = Some(p.into());
+    pub fn path<P: Into<PathBuf>>(&mut self, p: P) -> &mut AgateOptions {
```

Comment:
make it take `self` and return `Self`, so we can write like this 
```
let opts = AgateOptions::Default()
      .path(xxx)
      .num_memtables(xxx);
```

Also, I'm not sure if these setters are necessary. Maybe make all the fields public is enough?

Reply:
I'm not sure if this change is necessary. We could already write `AgateOptions::Default().path(xxx).open(xxx)` with current implementation. FYI, https://github.com/tikv/agatedb/blob/develop/src/db.rs#L317

Reply:
okay. Apart from that, are these setters necessary? Maybe make all the fields public is enough?

Reply:
In fact, I also prefer making all fields public (and check if they're valid when opening) cc @BusyJay